test/interval.jl

```diff
@@ -149,6 +177,58 @@ isinf(::TimeType) = false
         @test span(interval) == Hour(3)
     end
 
+    @testset "accessors - unbounded intervals" begin
+        unbounded_test_values = [
+            # one side unbounded with different types
+            (-10, nothing, 1, typeof(1), Open, Unbounded),
+            (nothing, 1, 0.01, Float64, Unbounded, Closed),
+            (nothing, 'z', 1, Char, Unbounded, Open),
+            (Date(2013, 2, 13), nothing, Day(1), Date, Closed, Unbounded),
+            (DateTime(2016, 8, 11, 0, 30), nothing, Millisecond(1), DateTime, Open, Unbounded),
+
+            #both sides unbounded different types
+            (nothing, nothing, 1, typeof(1), Unbounded, Unbounded),
+            (nothing, nothing, 0.01, Float64, Unbounded, Unbounded),
+            (nothing, nothing, 1, Char, Unbounded, Unbounded),
+            (nothing, nothing, Day(1), Day, Unbounded, Unbounded),
+            (nothing, nothing, Millisecond(1), DateTime, Unbounded, Unbounded),
+
+            # test adding eps() with unbounded
+            (-10.0, nothing, nothing, Float64, Open, Unbounded),
+            (nothing, 10.0, eps(Float64), Float64, Unbounded, Open),
+        ]
+        for (a, b, unit, T, L, R) in unbounded_test_values
+            interval = Interval{T,L,R}(a, b)
+
+            @test first(interval) == a
+            @test last(interval) == b
+
+            # The value we compare to min/max depends on if the bound is open/closed.
+            # we can determine if a bound is open/closed by checking if the edge in the interval
+            mi = if a === nothing
+                nothing
```

Comment:
I was originally thinking when using `minimum` on an left-unbounded interval you would get the minimum possible value in that interval. When taking into account the element type this could return `typemin(eltype(interval))`. There can definitely be some discussion on what we expect in this case